stdlib/bytes.ml

```diff
@@ -268,3 +268,34 @@ let lowercase s = map Char.lowercase s
 
 let capitalize s = apply1 Char.uppercase s
 let uncapitalize s = apply1 Char.lowercase s
+
+(** {6 Iterators} *)
+
+let to_iter s =
+  let next i =
+    if i = length s then Iter.Done
+    else
+      let x = get s i in
+      Iter.Yield (x, i+1)
+  in
+  Iter.Sequence (0, next)
+
+let to_iteri s =
+  let next i =
+    if i = length s then Iter.Done
+    else
+      let x = get s i in
+      Iter.Yield ((i,x), i+1)
+  in
+  Iter.Sequence (0, next)
+
+let of_list l =
+  let n = List.length l in
+  let b = make n '\000' in
+  List.iteri (fun i c -> set b i c) l;
+  b
+
+let of_iter g =
+  let l = List.of_iter g in
```

Comment:
Lists of characters are particularly inefficient.  I'd say: either provide a more efficient version (resizing buffers?) or leave it to the programmer to do it (e.g. mapping to arrays and then using `Bytes.init`).